islandora.module

```diff
@@ -186,7 +205,7 @@ function islandora_file_insert(FileInterface $file) {
  */
 function islandora_file_update(FileInterface $file) {
   // Exit early if unchanged.
-  if ($file->filehash['sha1'] == $file->original->filehash['sha1']) {
+  if ($file->filehash != NULL && $file->original->filehash != NULL && $file->filehash['sha1'] == $file->original->filehash['sha1']) {
```

Comment:
Comparison with `NULL` here seems... odd? Non-strict comparison... would be `TRUE` with `FALSE` and an empty string and the like, wouldn't it?

... would `isset()` tests make more sense?:

```suggestion
  if (isset($file->filehash) && isset($file->original->filehash) && $file->filehash['sha1'] == $file->original->filehash['sha1']) {
```